geode-assembly/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/StartLocatorCommandDUnitTest.java

```diff
@@ -0,0 +1,295 @@
+
+    final String expectedError = Launcher_Command_START_PID_FILE_ALREADY_EXISTS_ERROR_MESSAGE
+        .toString(LOCATOR_TERM_NAME, pidFile.getParentFile().getCanonicalPath(),
+            InetAddress.getLocalHost().getHostAddress() + "[0]");
+    final String expectedCause = "Caused by: "
+        + "org.apache.geode.internal.process.FileAlreadyExistsException: Pid file already exists: "
+        + pidFile.getCanonicalPath();
+
+    assertThat(result).contains(expectedError);
+    assertThat(result).contains(expectedCause);
+  }
+
+  @Test
+  public void testWithMissingGemFirePropertiesFile() throws IOException {
+    final String missingPropertiesPath = "/path/to/missing/gemfire.properties";
+    final String memberName = "testWithMissingGemFirePropertiesFile-locator";
+    final String expectedError =
+        MessageFormat.format(GEODE_0_PROPERTIES_1_NOT_FOUND_MESSAGE, "", missingPropertiesPath);
+
+    File workingDir = tempFolder.newFolder();
+
+    CommandStringBuilder command = new CommandStringBuilder(START_LOCATOR)
+        .addOption(START_LOCATOR__MEMBER_NAME, memberName)
+        .addOption(START_LOCATOR__DIR, workingDir.getAbsolutePath())
+        .addOption(START_LOCATOR__PROPERTIES, missingPropertiesPath);
+
+    GfshExecution execution = GfshScript.of(command.getCommandString()).execute(gfsh);
+    String result = execution.getOutputText();
+
+    assertThat(result).contains(expectedError);
+  }
+
+  @Test
+  public void testWithMissingGemFireSecurityPropertiesFile() throws IOException {
+    final String missingSecurityPropertiesPath = "/path/to/missing/gemfire-security.properties";
+    final String memberName = "testWithMissingGemFireSecurityPropertiesFile-locator";
+    final String expectedError = MessageFormat.format(GEODE_0_PROPERTIES_1_NOT_FOUND_MESSAGE,
+        "Security ", missingSecurityPropertiesPath);
+
+    File workingDir = tempFolder.newFolder();
+
+    CommandStringBuilder command = new CommandStringBuilder(START_LOCATOR)
+        .addOption(START_LOCATOR__MEMBER_NAME, memberName)
+        .addOption(START_LOCATOR__DIR, workingDir.getAbsolutePath())
+        .addOption(START_LOCATOR__SECURITY_PROPERTIES, missingSecurityPropertiesPath);
+
+    GfshExecution execution = GfshScript.of(command.getCommandString()).execute(gfsh);
+    String result = execution.getOutputText();
+
+    assertThat(result).contains(expectedError);
+  }
+
+  @Test
+  public void testWithUnavailablePort() throws IOException {
+    final Integer locatorPort = AvailablePortHelper.getRandomAvailableTCPPort();
+    final String memberName = "testWithUnavailablePort-locator";
+    final String unexpectedMessage = "[" + locatorPort + "] as locator is currently online.";
+    final String expectedMessage = "java.net.BindException: Network is unreachable; port ("
+        + locatorPort + ") is not available on localhost.";
+
+    Socket interferingProcess = new Socket();
```

Comment:
It's always a good idea to wrap resource-consuming code in a  try-finally block. In this case you could also do:
```
try (Socket interferingProcess = new Socket()) {
...
// no interferingProcess.close() required - it's automagic
}
```

Reply:
Should I be doing this in every test even though my JVMs will be shutdown in the end?